internal/elasticsearch/common.go

```diff
@@ -342,9 +345,6 @@ func newPodTemplateSpec(nodeName, clusterName, namespace string, node api.Elasti
 	proxyResourceRequirements := newESProxyResourceRequirements(node.ProxyResources, commonSpec.ProxyResources)
 
 	selectors := mergeSelectors(node.NodeSelector, commonSpec.NodeSelector)
-	// We want to make sure the pod ends up allocated on linux node. Thus we make sure the
-	// linux node selectors is always present. See LOG-411
-	selectors = utils.EnsureLinuxNodeSelector(selectors)
```

Comment:
did we keep the logic that is contained in this function and move it somewhere else that still gets used in this execution path?

Reply:
yes this logic moved to `internal/manifests/pod` in `newPodSpec`: https://github.com/periklis/elasticsearch-operator/blob/refactor-pkg-struct/internal/manifests/pod/build.go#L26

This logic was used three times across the `elasticsearch`, `kibana` and `indexmanagement` packages.